metronome/checkpointing/service/src/io/iohk/metronome/checkpointing/service/CheckpointingService.scala

```diff
@@ -141,16 +154,25 @@ class CheckpointingService[F[_]: Sync, N](
     }).flatten
   }
 
-  /** Computes a new ledger from the `block` and saves it in the ledger tree only if
-    * a parent state exists.
+  //TODO: PM-3107
+  def projectMempool: F[Mempool] =
+    mempoolRef.get
+
+  /** Saves a new ledger in the tree only if a parent state exists
+    * Depending on the `precomputed` parameter `ledger` is regarded as:
+    * - a precomputed ledger to be saved directly
+    * - the parent ledger from which a new ledger will be computed and saved
     *
     * Because we're only adding to the tree no locking around it is necessary
     */
-  private def updateLedgerByBlock(
-      oldLedger: Ledger,
-      block: Block
+  private def updateLedgerTree(
+      ledger: Ledger,
+      block: Block,
+      //TODO: I don't like this
+      precomputed: Boolean = false
   ): F[Ledger] = {
-    val newLedger = oldLedger.update(block.body.transactions)
+    val newLedger =
+      if (precomputed) ledger else ledger.update(block.body.transactions)
```

Comment:
I would not have guessed that `precomputed` means the ledger is for the block, otherwise it's for the parent. Maybe a separate method, or an `Either` (left=parent, right=child ledger)?

Reply:
Yeah, that sucked, fixed